solutions/src/main/java/ru/sberbank/school/task02/Calculator.java

```diff
@@ -48,36 +47,24 @@ public BigDecimal convert(ClientOperation operation, Symbol symbol, BigDecimal a
             throw new EmptyQuoteException("No quotes!");
         }
 
-        BigDecimal volume = BigDecimal.valueOf(0);
         Quote currentQuote = null;
 
         for (Quote q : quotes) {
-            if (volumeCompare(amount, q.getVolumeSize()) < 0) {
-                if (volumeCompare(volume, q.getVolumeSize()) > 0 || volume.compareTo(BigDecimal.ZERO) == 0) {
-                    volume = q.getVolumeSize();
+            if (amount.compareTo(q.getVolumeSize()) < 0 || q.isInfinity()) {
+                if (currentQuote == null || currentQuote.getVolumeSize().compareTo(q.getVolumeSize()) < 0
+                || q.isInfinity()) {
```

Comment:
&#39;if&#39; child has incorrect indentation level 16, expected level should be 20.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a1712913152>*